bundles/org.openhab.ui/web/src/assets/definitions/blockly/blocks-dateoffsets.js

```diff
@@ -173,14 +618,193 @@ export default function (f7) {
+        .appendField('and')
+        .setCheck('ZonedDateTime')
+      this.appendDummyInput()
+        .appendField('use')
+        .appendField(new Blockly.FieldDropdown([['date and time', 'dateandtime'], ['date', 'date'], ['time', 'time']]), 'dateComparison')
+
+      this.setOutput(true, 'Boolean')
+      this.setInputsInline(true)
+      this.setColour('%{BKY_LOGIC_HUE}')
+      this.setTooltip('Compares two ZonedDateTimes with each other')
+      this.setHelpUrl('https://www.openhab.org/docs/configuration/blockly/rules-blockly-date-handling.html#get-string-representation-of-date')
+    }
+  }
+
+  /*
+  * Returns a string representation of an ephemeris date
+  * Code part
+  */
+  Blockly.JavaScript['oh_zdt_between'] = function (block) {
+    let zdtCompare = addDateComparisonSupport()
+    let zdtOne = Blockly.JavaScript.valueToCode(block, 'zdtOne', Blockly.JavaScript.ORDER_ATOMIC)
+    let zdtTwo = Blockly.JavaScript.valueToCode(block, 'zdtTwo', Blockly.JavaScript.ORDER_ATOMIC)
+    let zdtThree = Blockly.JavaScript.valueToCode(block, 'zdtThree', Blockly.JavaScript.ORDER_ATOMIC)
+    let dateComparison = block.getFieldValue('dateComparison')
+
+    let codeLow = `${zdtCompare}(${zdtTwo}, ${zdtOne}, 'beforeEqual', 'nanos', '${dateComparison}')`
+    let codeHigh = `${zdtCompare}(${zdtOne}, ${zdtThree}, 'beforeEqual', 'nanos', '${dateComparison}')`
+    let code = `(${codeLow} && ${codeHigh})`
+    return [code, Blockly.JavaScript.ORDER_NONE]
+  }
+
+  /*
+  * Returns a temporal part of a zoned date time  as a Number
+  * Blockly part
+  */
+
+  Blockly.Blocks['oh_get_zdt_part'] = {
+    init: function () {
+      this.appendDummyInput()
+        .appendField(new Blockly.FieldDropdown([['year', 'getYear'], ['month', 'getMonthValue'], ['day', 'getDayOfMonth'], ['hour', 'getHour'], ['minute', 'getMinute'], ['second', 'getSecond'], ['milli', 'getMilli'], ['micro', 'getMicro'], ['nano', 'getNano']]), 'temporalPart')
+        .appendField('of')
+      this.appendValueInput('date')
+        .setCheck('ZonedDateTime')
+      this.setInputsInline(true)
+      this.setOutput(true, 'ZonedDateTime')
+      this.setColour(230)
+      this.setTooltip('Returns the selected field as Number.')
+      this.setHelpUrl('https://www.openhab.org/docs/configuration/blockly/rules-blockly-date-handling.html')
+    }
+  }
+
+  /*
+  * Returns a temporal part of a zoned date time as a Number
+  * Code part
+  */
+  Blockly.JavaScript['oh_get_zdt_part'] = function (block) {
+    let getZdtComponent = addGetZdtComponent()
+    let chrono = addChrono()
+    let temporalPart = block.getFieldValue('temporalPart')
+    switch (temporalPart) {
+      case 'getMilli':
+        temporalPart = `getLong(${chrono}.MILLI_OF_SECOND)`
+        break
+      case 'getMicro':
+        temporalPart = `getLong(${chrono}.MICRO_OF_SECOND) % 1000`
+        break
+      case 'getNano':
+        temporalPart = `getLong(${chrono}.NANO_OF_SECOND) % 1000`
+        break
```

Comment:
I didn't quite understand why the `% 1000` for micro & nano. For nano I seem to always get 0.
In fact I wonder if anything below a millisecond is really necessary, who would ever need them?

Reply:
First of all I didn't want to deviate from deibich's implementation as he had implemented nanos (and yes, you could of course question whether nanos are really necessary). I then went further because millis and nanos where there but micros was missing, so for consistency reasons I have added them. 

Regarding your note I checked the behaviour with

<img width="575" alt="image" src="https://user-images.githubusercontent.com/5937600/170963587-f9d56539-d4e5-4d98-bd24-a208be129688.png">

which results into 

>2022-05-30T11:34:26.855407+02:00[Europe/Berlin]
>2022-05-30T11:34:26.855407123+02:00[Europe/Berlin]
>123

So, you are right that now does not include Nanos itself, so we could have omitted them. Sure, you can set and add and subtract them but the question really is how much sense that makes.

Would you vote for removing nanos everywhere? I would then remove them with another PR.